Various changes (Inc alts part 2).

samuelgrant opened the pull request:
Changes include but not limited to: 
- Improved URL Schemes make things more standard
- Moved middleware to a `/middleware` directory and excluded the auth routes from the middlware auth check.
- Fixed the failure redirects in some of the admin controllers
- Add an NPM run script to minify a decent chunk of javascript. -- This cannot be used on production the minifed code must be in source control to work. **Will replace my custom solution with webpack later**.
- Removed some old static html files
- Cleaned up the list of requested scopes in Oauth to exclude scopes that aren't used.
- Various UX changes for FCs including feedback that indicates why an invite failed.
- AJAX polling for pilots on the waitlist page.

mlohstroh commented on Procfile:
```diff
@@ -0,0 +1 @@
+web: node index.js
```
Just a note for the future, we might need to make the build the JS and commit to the repo since we aren't running the minify up on heroku anymore.

mlohstroh commented on controllers/waitlistController.js:
```diff
@@ -46,37 +37,43 @@ exports.signup = function(req, res){
-            res.redirect(`/`);
+        var promise = new Promise(function(resolve, reject) {           
+            if(req.params.type === "main"){
+                waitlistMain = {
```
When using promises, instead of setting instance variables in a callback, we should really return something inside the resolve.

mlohstroh commented on controllers/waitlistController.js:
```diff
@@ -141,13 +138,176 @@ exports.clearWaitlist = function(req, res) {
+}
+
+module.getQueuePos = function(characterID, cb){
+    require('../dbHandler.js').db.collection('waitlist').find().sort({ signup: 1 }).toArray(function(err, docs){
```
Can this be moved to a model?

mlohstroh replied:
Feels like a code smell having an inline require like this.

mlohstroh commented on dbHandler.js:
```diff
@@ -3,26 +3,28 @@ const mongo = require('mongodb').MongoClient;
-			}
+                dbService.db = client.db(MONGO_DB);
+            }
+            
```
It looks like I very much screwed up the tabs versus spaces here. Can we fix this? I'll do it if need be.

mlohstroh commented on models/waitlist.js:
```diff
@@ -39,12 +39,12 @@ module.exports = function (setup) {
+                //         var disciplinary = true;
+                //         break;
+                //     }
+                // }
```
If it's commented, and probably won't be used within the next commit, lets remove it.

samuelgrant replied:
This feature needs to be working, currently I don't have the ability to do dev work on my computer as I don't have .env injection working. So, this will stay commented and will be resolved asap. 

Reference #2

mlohstroh commented on models/waitlist.js:
```diff
@@ -75,14 +75,18 @@ module.exports = function (setup) {
+                    db.remove({"waitlistMain.characterID": Number(pilot.characterID)}, function (err) {
+                        if (err) {
+                            if (err) log.error("waitlist.remove: Error for db.remove", { err, 'character ID': characterID });
+                            cb({"class": "error", "title": "Woops!", "message":"We could not remove you from the waitlist!"});
```
There is a slight bug that could happen here. If we have multiple pilots that get requested and say that the DB is unavailable, you will have multiple callbacks of error most likely on an invalid request object.

A way to solve this is to wrap all of this into a promise, and wait for all of them, and then call the callback once after all the promises finish.

mlohstroh commented on models/waitlist.js:
```diff
@@ -123,35 +127,11 @@ module.exports = function (setup) {
 				cb(false)
 			} else {
-				cb(true);
+                cb(doc.signup);
```
Consider a !! here.

samuelgrant commented:
Changes include but not limited to: 
- Improved URL Schemes make things more standard
- Moved middleware to a `/middleware` directory and excluded the auth routes from the middlware auth check.
- Fixed the failure redirects in some of the admin controllers
- Add an NPM run script to minify a decent chunk of javascript. -- This cannot be used on production the minifed code must be in source control to work. **Will replace my custom solution with webpack later**.
- Removed some old static html files
- Cleaned up the list of requested scopes in Oauth to exclude scopes that aren't used.
- Various UX changes for FCs including feedback that indicates why an invite failed.
- AJAX polling for pilots on the waitlist page.

Resolves #2
